webapi/src/main/scala/org/knora/webapi/responders/v2/ResourcesResponderV2.scala

```diff
@@ -589,8 +587,65 @@ final case class ResourcesResponderV2(
     targetSchema: ApiV2Schema,
     schemaOptions: Set[Rendering],
     requestingUser: User,
+  ): Task[ReadResourcesSequenceV2] =
+
+    for {
+
+      apiResponse <- getResources(
+                       resourceIris,
+                       propertyIri,
+                       valueUuid,
+                       versionDate,
+                       withDeleted,
+                       showDeletedValues,
+                       targetSchema,
+                       schemaOptions,
+                       requestingUser,
+                     )
+
+      // BR: Deleted resources are to be replaced with DeletedResource.
+      deletedResourcesReplaced =
+        apiResponse.resources.map { resource =>
+          resource.deletionInfo match {
+            // Resource deleted -> return DeletedResource instead
+            case Some(_) => resource.asDeletedResource(versionDate)
+            // Resource not deleted -> return resource
+            case None =>
+              // deleted values should be shown -> resource can be returned
+              if (showDeletedValues) resource
+              // deleted Values should not be shown -> replace them with generic DeletedValue
+              else resource.withDeletedValues(versionDate)
+          }
+        }
+      responseWithDeletedResourcesReplaced = apiResponse.copy(resources = deletedResourcesReplaced)
+    } yield responseWithDeletedResourcesReplaced
+
+  /**
+   * Get one or several resources and return them as a sequence.
+   *
+   * @param resourceIris      the IRIs of the resources to be queried.
+   * @param propertyIri       if defined, requests only the values of the specified explicit property.
+   * @param valueUuid         if defined, requests only the value with the specified UUID.
+   * @param versionDate       if defined, requests the state of the resources at the specified time in the past.
+   * @param withDeleted       if defined, indicates if the deleted resource and values should be returned or not.
+   * @param showDeletedValues if false, deleted values will be shown as DeletedValue
+   * @param targetSchema      the target API schema.
+   * @param schemaOptions     the schema options submitted with the request.
+   * @param requestingUser    the user making the request.
+   * @return a [[ReadResourcesSequenceV2]].
+   */
+  def getResources(
```

Comment:
I find it confusing that we have now two `getResources` methods which produce two different results, how about renaming the other `getResourcesWithDeletedResource` to convey its intent?